dockerfiles/executor_image/Dockerfile

```diff
@@ -18,26 +24,36 @@ RUN ARCH=$(dpkg --print-architecture) && \
     echo "$SHASUM /tini" | sha256sum -c && \
     chmod +x /tini
 
-RUN install -m 0755 -d /etc/apt/keyrings && \
+RUN DOCKER_VERSION="5:26.1.4-1~debian.12~bookworm" && \
+    CONTAINERD_DEB_VERSION="1.7.19-1" && \
+    DOCKER_BUILDX_VERSION="0.15.1-1~debian.12~bookworm" && \
+    install -m 0755 -d /etc/apt/keyrings && \
     curl -fsSL https://download.docker.com/linux/debian/gpg -o /etc/apt/keyrings/docker.asc && \
     chmod a+r /etc/apt/keyrings/docker.asc && \
     echo \
       "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/debian \
       $(. /etc/os-release && echo "$VERSION_CODENAME") stable" | \
     tee /etc/apt/sources.list.d/docker.list > /dev/null && \
-    apt-get update && apt-get install -y docker-ce docker-ce-cli containerd.io && \
-    rm -rf /var/lib/apt/lists/* && apt-get clean
-
-# Install skopeo and umoci which we use to unpack OCI images when we're not using docker.
-# Install iproute2 ("ip" command) to configure networking on host.
-RUN apt-get update && \
-    apt-get install -y skopeo umoci iproute2 amazon-ecr-credential-helper && \
-    rm -rf /var/lib/apt/lists/* && apt-get clean
-
-# Install gVisor
-#RUN curl -fsSL https://gvisor.dev/archive.key | gpg --dearmor -o /usr/share/keyrings/gvisor-archive-keyring.gpg && echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/gvisor-archive-keyring.gpg] https://storage.googleapis.com/gvisor/releases release main" >>  /etc/apt/sources.list.d/gvisor.list && apt-get update && apt-get install -y runsc
-# ALTERNATIVE non-apt method:
-#RUN ARCH=$(uname -m); URL=https://storage.googleapis.com/gvisor/releases/release/latest/${ARCH}; curl -LO ${URL}/runsc && curl -LO ${URL}/runsc.sha512 && sha512sum -c runsc.sha512 && rm -f *.sha512 && chmod a+rx runsc && mv runsc /usr/local/bin
+    apt-get update && apt-get install -y \
+    docker-ce=${DOCKER_VERSION} \
+    docker-ce-cli=${DOCKER_VERSION} \
+    containerd.io=${CONTAINERD_DEB_VERSION} \
+    docker-buildx-plugin=${DOCKER_BUILDX_VERSION} && \
+    apt-get autoremove -y && \
+    rm -rf /var/lib/apt/lists/* && apt-get clean && \
+    rm \
+    # keep these in sync with the binaries we are manually adding
```

Comment:
could you add some notes to the README about how to maintain this step? i.e. when upgrading docker, what process should we follow to add/remove deps from this list in case the deps have changed?

Reply:
Done!